extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/RuntimeBeanBuildItem.java

```diff
@@ -102,8 +113,18 @@ public Builder setRemovable(boolean removable) {
             return this;
         }
 
+        public Builder setSupplier(Supplier<Object> supplier) {
+            this.supplier = Objects.requireNonNull(supplier);
+            return this;
+        }
+
+        public Builder setRuntimeValue(RuntimeValue<?> runtimeValue) {
+            this.value = Objects.requireNonNull(runtimeValue);
+            return this;
+        }
+
         public RuntimeBeanBuildItem build() {
-            return new RuntimeBeanBuildItem(scope, type, supplier, qualifiers, removable);
+            return new RuntimeBeanBuildItem(scope, type, supplier, qualifiers, removable, value);
```

Comment:
I think I would have put `supplier` together with `value` at the end of the constructor.

Reply:
No one will use this constructor directly so I don't think it really matters.